packages/pyo3/src/mediator.rs

```diff
@@ -188,29 +188,36 @@ fn communities_to_clustering(
     network: &LabeledNetwork<String>,
     communities: HashMap<String, usize>,
 ) -> Result<Clustering, PyLeidenError> {
-    // if node not found in internal network, bounce
-    // if max(communities) > len(set(communities)), collapse result
-    // if all nodes are mapped, cool
-    // if not all nodes are mapped, generate integers for each new value
-
-    let mut clustering: Clustering = Clustering::as_self_clusters(network.num_nodes());
-
-    let mut all_communities: HashSet<usize> = HashSet::new();
+    // best effort for using a provided community mapping and keeping generated singleton ids
+    // otherwise
+
+    let mut clustering: Clustering = match communities.values().max() {
+        Some(max_community_id) => {
+            // if we have values, we have the max community id, which means we'll actually create
+            // a cluster mapping where every node gets a singleton cluster starting after the actual
+            // last max cluster.
+            let mut cluster_mapping: Vec<usize> = Vec::with_capacity(network.num_nodes());
+            let start: usize = max_community_id + 1;
+            let end: usize = start + network.num_nodes();
+            cluster_mapping.extend(start..end);
+            Clustering::as_defined(cluster_mapping, end)
+        }
+        None => Clustering.as_self_clusters(network.num_nodes()),
+    };
 
     for (node, community) in communities {
-        all_communities.insert(community);
-        let mapped_node: CompactNodeId = network
-            .compact_id_for(node)
-            .ok_or(PyLeidenError::InvalidCommunityMappingError)?;
-        clustering
-            .update_cluster_at(mapped_node, community)
-            .map_err(|_| PyLeidenError::InvalidCommunityMappingError)?;
+        let mapping: Option<CompactNodeId> = network.compact_id_for(node);
+        if mapping.is_some() {
+            let compact_node_id: CompactNodeId = mapping.unwrap();
+            clustering
```

Comment:
the only way this error should occur is if the clustering we generated didn't match up with the size of the network we have, which would be a bug that any unit test should find in pretty short order

Reply:
are you able to capture the details in ClusterIndexingError? (ie which node and community had this issue?)